veg2hab/vegkartering.py

```diff
@@ -508,6 +509,45 @@ def finalize_final_format(gdf: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
     return gdf[new_columns]
 
 
+def _combineer_twee_geodataframes(
+    lage_prio: gpd.GeoDataFrame, hoge_prio: gpd.GeoDataFrame
+) -> gpd.GeoDataFrame:
+    """
+    Geometrische-operaties-hulpfunctie voor combine_karteringen
+
+    Combineert twee geodataframes op basis van de geometrie; bij overlaps
+    wordt de geometrie van lage_prio bijgesneden tot de overlap met hoge_prio
+
+    De data van alle vlakken blijft onveranderd (zolang ze blijven bestaan)
+
+                +-------+
+    Lage_prio:  |   A   |
+                +-------+
+                    +-------+
+    Hoge_prio:      |   B   |
+                    +-------+
+                +---+-------+
+    Resultaat:  | A |   B   |
+                +---+-------+
+    """
+    # Snij de geometrie van lage_prio bij tot de overlap met hoge_prio
+    new_gdf = gpd.overlay(lage_prio, hoge_prio, how="difference")
+
+    # Voeg de geometrie van hoge_prio toe
+    new_gdf = new_gdf.append(hoge_prio)
+
+    # Exploden eventuele multipolygons naar polygons
+    new_gdf = new_gdf.reset_index(drop=True).explode(ignore_index=True)
```

Comment:
Supporten wij alleen gewone geometrieen tot nu toe?

Reply:
ik ben het hier overigens wel mee eens.

Reply:
Ik denk het wel. Er stond ieg in de ticket dat er gesplitst moest worden.